### PR TITLE
feat(button): implement forceCompact on buttons

### PR DESCRIPTION
**Pull request by piofinn**
## 📥 Proposed changes

Implement ability to force compact mode on buttons.

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

## 💬 Further comments

affects: @fremtind/jkl-button-react, @fremtind/jkl-button

### CODE REVIEW

File: packages/button-react/example/index.tsx [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+                Svar senere
+            </SecondaryButton>
         </div>
-        <div style={{ margin: "20px" }}>
-            <TertiaryButton onClick={onClick}>Tertiary</TertiaryButton>
+        <div className="side-by-side">
+            <TertiaryButton onClick={onClick}>Avbryt</TertiaryButton>
+            <TertiaryButton forceCompact onClick={onClick}>
+                Avbryt
+            </TertiaryButton>
```
Comment (lfbergee) [inline packages/button-react/example/index.tsx]: If you view this on a mobile-device, it looks the same. That might not be an issue, but the demo kinda don't work in that case.

Is there any case, besides a demo like this to force into large?

Comment (piofinn) [reply packages/button-react/example/index.tsx]: Yeah, I see your point! It does still demonstrate the functionality, but it's impossible to see the difference 🤔 It looks like we're heading towards keeping the big size on mobile, so I see no point in implementing a `forceLarge` functionality per now…

File: packages/button-react/src/Button.test.tsx [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+            </>,
+        );
+
+        const primaryButton = getByText("primary");
+        const secondaryButton = getByText("secondary");
+        const tertiaryButton = getByText("tertiary");
+
+        expect(primaryButton).toHaveClass("jkl-button--primary");
+        expect(secondaryButton).toHaveClass("jkl-button--secondary");
+        expect(tertiaryButton).toHaveClass("jkl-button--tertiary");
```
Comment (lfbergee) [inline packages/button-react/src/Button.test.tsx]: I don't like multiple test in a test.

I'd prefer to split this into three test. Then each will report as its own test.

```
[
        { name: "Primary", component: PrimaryButton },
        { name: "Secondary", component: SecondaryButton },
        { name: "Tertiary", component: TertiaryButton },
    ].map((buttonVariant) => {
        it(`should work for ${buttonVariant}`, () => {
            const C = buttonVariant.component;

            const { getByText } = render(<C onClick={() => {}}>{buttonVariant.name}</C>);
            expect(getByText(buttonVariant.name)).toHaveClass(`jkl-button--${buttonVariant.name.toLowerCase()}`);
        });
    });
```

Comment (piofinn) [reply packages/button-react/src/Button.test.tsx]: Ooh, I like this!

File: packages/button-react/example/index.tsx (hunk 1 above)
Comment (lfbergee) [reply packages/button-react/example/index.tsx]: For sure, we should not implement forceLarge just for test/demo purposes.

**Pull request merged**